### DISCUSSION TITLE
[HowTo]: Mount Additional Storage in OMV LXC on Proxmox

### DISCUSSION THREAD

**Discussion by MickLesk** [community-scripts/ProxmoxVE#175] [Guides]
# Mounting Additional Storage in OMV LXC on Proxmox

This guide explains how to mount additional storage in an OpenMediaVault (OMV) container running on Proxmox Virtual Environment (PVE).

## Prerequisites
- Proxmox Virtual Environment (PVE) host
- Existing OMV LXC container

### 1. Create OMV LXC Container
Create an OMV LXC container on your Proxmox host using the corresponding script. Ensure the container is set to "Privileged" during creation.

### 2. Add Storage Mount Point
1. Stop the OMV container.
2. In the Proxmox shell, add a mount point for additional storage:
   ```
   pct set <CT_ID> -mp0 local-lvm:10,mp=/data
   ```
   This will create a 10GB storage mount point at `/data`.

### 3. Find Block Device IDs
1. Run `lvdisplay` in the Proxmox shell.
2. Look for the Logical Volume like `/dev/pve/vm-<CT_ID>-disk-X`.
3. Note the "Block device" numbers, e.g., 253 9.

![image](https://github.com/user-attachments/assets/332b54ee-38ed-470d-85fb-5d8adaeffa19)

### 4. Create Mount Hook Script
1. Create the mount hook script file:
   ```
   echo '#!/bin/sh' > /var/lib/lxc/<CT_ID>/mount-hook.sh
   echo 'mknod -m 777 ${LXC_ROOTFS_MOUNT}/dev/sda1 b 253 9' >> /var/lib/lxc/<CT_ID>/mount-hook.sh
   ```
   Replace `253 9` with the block device numbers you found in the previous step.
2. Make the script executable:
   ```
   chmod a+x /var/lib/lxc/<CT_ID>/mount-hook.sh
   ```

### 5. Add Hook to LXC Config
Edit the LXC config file:
```
echo 'lxc.hook.autodev: /var/lib/lxc/<CT_ID>/mount-hook.sh' >> /etc/pve/lxc/<CT_ID>.conf
```

### 6. Start the OMV Container
Start the OMV container.

### 7. Mount the Storage
1. In the OMV web interface, go to "Storage" -> "File Systems".
2. Click "Mount existing file system" and add `/dev/sda1`.

At this point, the additional storage should be available in the OMV interface. You can then create shares, set permissions, and use the storage as needed.

Please note that some users reported issues with this method, particularly around the block device numbers changing after a reboot. In such cases, you may need to modify the mount hook script to use the LVM volume UUID instead of the block device numbers.

Original-Thread: https://github.com/tteck/Proxmox/discussions/1075

**Comment by Koliera** [comment]
Question - would this work with the scenario of just adding an existing hdd to an LXC? Such as adding an existing HDD to a jellyfin server LXC?
For VMs I usually just use the following - https://i12bretro.github.io/tutorials/0653.html
But unsure how to do the same to a LXC.

**Reply by NAM1025** [reply]
This does actually seem to work.

I found my block device info via `lsblk` for a drive that I was using in OMV in a VM (via disk passthrough) and just plugged it in...  It recognized the drive in its entirety on OMV in the new LXC I setup.

Edit:
Found a script in the old thread linked in the OP. 

I modified it to work with `lsblk` so that you could use the UUID to find the correct drive if you wanted to set it up this way.

```sh
#!/bin/sh
#MAJ_MIN=`lsblk -yn -o MAJ_MIN --filter "UUID == 'b581977e-ea73-48b9-a894-a8697d8c6d15'" | sed "s/:/ /g"`  # Needs a newer version of lsblk
MAJ_MIN=`lsblk -n -o "MAJ:MIN,UUID" | grep "b581977e-ea73-48b9-a894-a8697d8c6d15" -m 1 | sed "s/\([0-9]\+\)\:\([0-9]\+\).*/\\1 \\2/g"`

mknod -m 777 ${LXC_ROOTFS_MOUNT}/dev/sda1 b $MAJ_MIN
```

**Reply by ChrizZz90** [reply]
do I understand this correctly, that with using existing hdds I can skip points 1-3, starting directly with the 4th step and using this update with the UUID?

//edit: Yes, just starting with step 4 and use NAM1025 adjusted script with your own hdd UUID

**Reply by adriaanos** [reply]
Can confirm that the above by ChrizZz90 works.

My only problem now is once I create the share and want to write to any files that have already been created I get a permissions error. I have everything chown by 1005 so wondering how I do the same for lets say a random "samba" user here?

**Comment by Koliera** [comment]
I do wonder if the above might be created as a script similar to the tailscale one - as in you run the script, select lxc, select the hdd option you want to use from a generated list, and then done. Is that an option?

**Comment by n8ur** [comment]
I'm a little confused by a couple of things in the original post:

1.  The prereq includes an existing OMV LXC container, but the first step is to create an OMV LXC.  Do I make a second one, or is the prerequisite wrong?
2.  On my system, doing lvdisplay in the proxmox shell shows three volumes: root, swap, and data.  There's nothing with a name referencing a VM or a container.  Where does that "vm-103-disk-1" reference come from, and what do I do in my situation?
3. I have a separate ZFS RAIDZ2 pool created which is where I'd like to put my OMV storage.  How would I access that?

Thanks!

**Comment by blablazzz** [comment]
Could we have script for OpenMediaVault in VM?

**Reply by Koliera** [reply]
I'd say it should go beyond the OMV VM, because many different scripts would benefit from linking into the hdd.

**Comment by Da-Hauge** [comment]
Does this only work with LVM drives? I would like to use my ZFS pool, but I don't see the additional storage with `lvdisplay`

**Comment by VasiaKit** [comment]
<CT_ID> — це ідентифікатор контейнера (Container ID)
Як приклад використано диск /dev/sdc та розділ /dev/sdc1

Рішення: Дати OMV доступ до блочних пристроїв
Потрібно змінити конфігурацію контейнера OMV:

Зупиніть контейнер

`pct stop <CT_ID>`

Відкрийте конфігураційний файл контейнера

`nano /etc/pve/lxc/<CT_ID>.conf`

Додайте або змініть ці параметри

```
lxc.cgroup2.devices.allow = b 8:* rwm
lxc.mount.entry = /dev/sdc dev/sdc none bind,optional,create=file
lxc.mount.entry = /dev/sdc1 dev/sdc1 none bind,optional,create=file
lxc.apparmor.profile = unconfined
lxc.cap.drop =
lxc.cap.add = SYS_ADMIN SYS_RAWIO

```
Збережіть і закрийте файл (CTRL+X, Y, Enter).

Запустіть контейнер

`pct start <CT_ID>`

**Reply by nyakojiru** [reply]
<CT_ID> is the Container ID
The example uses the /dev/sdc disk and the /dev/sdc1 partition

Solution: Give OMV access to the block devices
You need to reconfigure the OMV container:

Stop the container

`pct stop <CT_ID>`

Open the container configuration file

`nano /etc/pve/lxc/<CT_ID>.conf`

Add or change these parameters

```
lxc.cgroup2.devices.allow = b 8:* rwm
lxc.mount.entry = /dev/sdc dev/sdc none bind,optional,create=file
lxc.mount.entry = /dev/sdc1 dev/sdc1 none bind,optional,create=file
lxc.apparmor.profile = unconfined
lxc.cap.drop =
lxc.cap.add = SYS_ADMIN SYS_RAWIO
```


Save and close the file (CTRL+X, Y, Enter).

Start the container

`pct start <CT_ID>`

**Comment by alconino** [comment]
When will be avaible under VM ?

**Comment by alconino** [comment]
ok, thanks, i reset the proxmox server and install again and it seems
everything is ok.

Thanks anyway


escribió:

**Reply by kendikaku** [reply]
> ok, thanks, i reset the proxmox server and install again and it seems everything is ok. Thanks anyway *David Alcón *| ***@***.*** | +34 679 82 07 01 El mié, 9 jul 2025 a las 12:52, Patricio ***@***.***>) escribió:
> […](#)
> <CT_ID> is the Container ID The example uses the /dev/sdc disk and the /dev/sdc1 partition Solution: Give OMV access to the block devices You need to reconfigure the OMV container: Stop the container pct stop <CT_ID> Open the container configuration file nano /etc/pve/lxc/<CT_ID>.conf Add or change these parameters lxc.cgroup2.devices.allow = b 8:* rwm lxc.mount.entry = /dev/sdc dev/sdc none bind,optional,create=file lxc.mount.entry = /dev/sdc1 dev/sdc1 none bind,optional,create=file lxc.apparmor.profile = unconfined lxc.cap.drop = lxc.cap.add = SYS_ADMIN SYS_RAWIO Save and close the file (CTRL+X, Y, Enter). Start the container pct start <CT_ID> — Reply to this email directly, view it on GitHub <[#175 (reply in thread)](https://github.com/community-scripts/ProxmoxVE/discussions/175#discussioncomment-13707930)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/BERFINMOFCFS2AUDFH6AGY33HTX67AVCNFSM6AAAAABRR2MNVOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTGNZQG44TGMA> . You are receiving this because you commented.Message ID: ***@***.*** .com>

How did you do it? Could you pass it on to me?

**Comment by alconino** [comment]
I backed up the VMs or LXCs to an external HDD or USB, reinstall Proxmox
from scratch and restore the VMs and LXCs.

*David Alcón *|  ***@***.*** | +34 679 82 07 01

**Comment by samcux** [comment]
## Setup

- **Host:** Proxmox VE 9.1.1 (ZFS)
- **Guest:** Debian 12 LXC
- **In guest:** OpenMediaVault + Docker
- **Data disk:** ZFS zvol on host, passed into LXC as block device `/dev/zd32`

`/etc/pve/lxc/104.conf` (relevant):

```ini
rootfs: local-zfs:subvol-104-disk-0,size=4G

lxc.cgroup2.devices.allow: b 230:* rwm
lxc.mount.entry: /dev/zd32 dev/zd32 none bind,optional,create=file                  #OMV needs same device name as on host

# for OMV to see UUIDs
lxc.mount.entry: /dev/disk dev/disk none bind,optional,create=dir

lxc.apparmor.profile: unconfined
````

In the container:

```bash
ls -l /dev/disk/by-uuid
f5b1c01b-217b-4bee-8734-b4d341b98600 -> ../../zd32
```



`/etc/fstab`:

```fstab
UUID=f5b1c01b-217b-4bee-8734-b4d341b98600  /srv/dev-disk-by-uuid-f5b1c01b-217b-4bee-8734-b4d341b98600  btrfs  defaults  0  2
```


<img width="1586" height="296" alt="grafik" src="https://github.com/user-attachments/assets/90698d0c-f356-4a10-ad86-374e2281493c" />





## Problem

* On **boot**, the BTRFS filesystem is **not** mounted.

* After boot, running:

  ```bash
  mount -a
  ```

  mounts it immediately and without error:

  ```bash
  /dev/zd32 on /srv/dev-disk-by-uuid-f5b1c01b-217b-4bee-8734-b4d341b98600 type btrfs (...)
  ```

* OMV/monit log:

  ```text
  filesystem '/srv/dev-disk-by-uuid-f5b1c01b-217b-4bee-8734-b4d341b98600' not mounted
  ```

  which is correct at boot time.

It looks like a **timing/race issue**: when systemd processes fstab, `/dev/zd32` inside the LXC is not ready yet; later it exists and `mount -a` works.

I also tried adding systemd options in fstab, e.g.:

```fstab
UUID=f5b1c01b-... /srv/dev-disk-by-uuid-... btrfs \
  defaults,x-systemd.requires=dev-zd32.device,x-systemd.after=dev-zd32.device,x-systemd.device-timeout=60 0 2
```

but the behaviour is the same: not mounted at boot, only manual `mount -a` works.

## Questions

* Is this expected behaviour for a passed-through block device (`/dev/zd32` zvol) in a LXC with systemd?
* Is there a recommended way to ensure this filesystem is mounted during boot inside the LXC (fstab vs. dedicated `.mount` unit vs. `systemd-udev-settle`, etc.)?
* Am I missing something obvious in the LXC config or fstab so that systemd treats `/dev/zd32` as a proper device dependency?

Any pointers appreciated.

**Reply by midi1996** [reply]
hello, I made my own .mount units, and it's working consistently across restarts. here is how you make one:

- run `systemd-escape -p --suffix=mount "/srv/mountPathByOMV"`, this will output the unit filename to use
- run `touch /etc/systemd/system/outputOfPrevCommand.mount`, this will create the unit file with the proper filename and escapes
- edit the unit with the following:
```
[Unit]
Description=Mount disk <name> # this is a description, put whatever you like
After=local-fs-pre.target

[Mount]
What=/dev/disk/by-uuid/UUID # path to the disk block
Where=/srv/OMVmountpoint # where OMV mounts the block
Type=btrfs # filesystem used, optional
Options=defaults,noatime,nofail # mount options

[Install]
WantedBy=multi-user.target
```

Do this for every mount point.

**Comment by tigaente** [comment]
I followed the guide here to the letter. However, after the OMV container starts back up and I want to create a filesystem (Storage -> Filesystem -> Mount existing), the drop down box doesn't show anything.

Any ideas?
